lib/money_oxr/rates_store.rb

```diff
@@ -84,7 +85,8 @@ def api_uri
     end
 
     def load_from_cache_path
-      load_json(File.read(cache_path))
+      json = File.read(cache_path)
+      load_json(json) unless json.empty?
```

Comment:
Why would it be empty?  If it's a valid use-case, please a unit test for this.

Reply:
Specs fail without this condition. File can be opened for writing, but not filled yet.

Reply:
Weird.  I tracked this down to strange behavior from webmock.  If you stub the body of a request with a string, it modifies and clears the string you provide it.  Thanks webmock.  I updated master.  Can you remove this code?

Reply:
Done. But specs still fail without this condition.

Reply:
Do you know why?  master branch is now passing without it.

Reply:
Yes, I know. I wrote it above, let me to explain more detail:

The value of this PR is file locks.

Error occurs when one thread (process) is reading file while another process is writing in this file.

Now we're locking file when want to read or write.

There is `with_cache_file_lock` method with block for it.

This (discussing) code is in `load_from_cache_path` method, which called from `with_cache_file_lock do` block, which yielding inside `File.open(cache_path, 'a') do |cache_io|` block.

So, after `File.open` (cache) file exists, but it's empty.